src/VisualStudio/Core/Def/Implementation/Venus/VenusCommandFilter.cs

```diff
@@ -61,49 +61,52 @@ public override int GetDataTipText(TextSpan[] pSpan, out string pbstrText)
                 return VSConstants.E_FAIL;
             }
 
-            // We need to map the TextSpan from the DataBuffer to our subject buffer. We'll
-            // only consider spans whose length matches our input span (which we expect to
-            // always be zero). This is to address the case where the position is on a seam
-            // and maps to multiple source spans.
-            // If there is not exactly one matching span, just return.
+            // We need to map the TextSpan from the DataBuffer to our subject buffer.
             var span = textViewModel.DataBuffer.CurrentSnapshot.GetSpan(pSpan[0]);
-            var spanLength = span.Length;
-            Debug.Assert(spanLength == 0, $"Expected zero length span (got '{spanLength}'.");
-            var subjectSpan = WpfTextView.BufferGraph.MapDownToBuffer(span, SpanTrackingMode.EdgeInclusive, _subjectBuffer)
-                                .SingleOrDefault(x => x.Length == spanLength);
+            var subjectSpans = WpfTextView.BufferGraph.MapDownToBuffer(span, SpanTrackingMode.EdgeInclusive, _subjectBuffer);
 
-            if (subjectSpan == default(SnapshotSpan))
+            // The following loop addresses the case where the position is on a seam and maps to multiple source spans.
+            // In these cases, we assume it's okay to return the first span that successfully returns a DataTip.
+            // It's most likely that either only one will succeed or both with fail.
+            var expectedSpanLength = span.Length;
+            foreach (var candidateSpan in subjectSpans)
             {
-                pbstrText = null;
-                return VSConstants.E_FAIL;
-            }
+                // First, we'll only consider spans whose length matches our input span. 
+                if (candidateSpan.Length != expectedSpanLength)
+                {
+                    continue;
+                }
 
-            pSpan[0] = subjectSpan.ToVsTextSpan();
+                // Next, we'll check to see if there is actually a DataTip for this candidate.
+                // If there is, we'll map this span back to the DataBuffer and return it.
+                pSpan[0] = candidateSpan.ToVsTextSpan();
```

Comment:
It looks like we're modifying the value even in the case where this method returns ```E_FAIL``` though. If that was the previous behavior, ok.